### DISCUSSION TITLE
expected `datafusion::arrow::record_batch::RecordBatch`, found `arrow::record_batch::RecordBatch`

### DISCUSSION THREAD

**Discussion by gianarb** [apache/datafusion#5045] [Q&A]
Hello!

My use case looks like this: I am reading an ipc stream and I would like to register record batches to datafusion, this is the snippet of the code I wrote:

```rust
// v is a ipc bytes buffer
let ctx = SessionContext::new();
let cursor = std::io::Cursor::new(v);
let ipc_reader = arrow::ipc::reader::StreamReader::try_new(cursor).unwrap();
let batch = ipc_reader.next().unwrap().unwrap();
ctx.register_batch("t", batch).unwrap();
```

`register_batch` fails with such error and I am not sure about how to move from a arrow::record_batch to a datafusion::arrow::record_batch or if I am doing all of this in the wrong way.

expected `datafusion::arrow::record_batch::RecordBatch`, found `arrow::record_batch::RecordBatch`

Thanks

**Comment by metesynnada** [comment]
You can use `datafusion::arrow::ipc::reader::StreamReader::try_new(cursor).unwrap()` while creating an ipc reader. It would solve the problem.

**Comment by tustvold** [comment]
> expected datafusion::arrow::record_batch::RecordBatch, found arrow::record_batch::RecordBatch

Is this a compilation error or a runtime error. If it is a compilation error you are likely mixing arrow versions, you need to use the same version of arrow as datafusion is. If you can share your Cargo.toml I can advise

**Comment by gianarb** [comment]
Hello! Sorry it took me that long to report back. You are rigth @tustvold it was a dependency conflict between datafusion and arrow
